src/ffi/ios/structs.rs

```diff
@@ -0,0 +1,177 @@
+use std::{
+    borrow::Cow,
+    collections::HashMap,
+    convert::TryFrom,
+    mem::{self, ManuallyDrop},
+};
+
+use anyhow::Context;
+use core_foundation::{
+    array::CFArray,
+    base::{FromVoid, TCFType},
+    data::{CFData, CFDataRef},
+    dictionary::{CFDictionary, CFDictionaryRef},
+    string::{CFString, CFStringRef},
+    url::{CFURLRef, CFURL},
+};
+use http::{
+    header::{HeaderName, HeaderValue},
+    Uri,
+};
+
+#[repr(C)]
+pub enum Method {
+    // TODO more methods!
+    Get,
+}
+
+impl From<Method> for http::Method {
+    fn from(method: Method) -> Self {
+        match method {
+            Method::Get => http::Method::GET,
+        }
+    }
+}
+
+#[repr(C)]
+pub struct Request {
+    method: Method,
+    url: CFURLRef,
+    /// CFDictionary<CFString, CFArray<CFData>>,
```

Comment:
Can this be removed?

Reply:
It can but it might really help anyone wanting to know the contained types (as `CFDictionaryRef` is untyped)

Reply:
Added comments